lib/standard/text/abstract_text.nit

```diff
@@ -330,6 +361,46 @@ abstract class Text
 		return if neg then -val else val
 	end
 
+	# Is `self` a valid integer ?
+	#
+	#     assert "0xFE46u8".is_num
+	#     assert "0b0100".is_num
+	#     assert "0o645".is_num
+	#     assert "897u8".is_num
+	fun is_num: Bool do
+		var prefix = get_numhead
+		var s = strip_numhead.strip_numext.remove_all('_')
+		if prefix != "" then
+			var c = prefix[1]
+			if c == 'x' or c == 'X' then return s.is_hex
+			if c == 'o' or c == 'O' then return s.is_oct
+			if c == 'b' or c == 'B' then return s.is_bin
+		end
+		return s.is_dec
+	end
+
+	# Is `self` is a properly formatted integer, returns the corresponding value
+	#
+	#     assert "0xFEu8".to_num  == 254u8
+	#     assert "0b10_10".to_num != 10u8
+	#
+	# REQUIRE: `self.is_num`
+	fun to_num: Numeric do
+		assert is_num
+		var s = remove_all('_')
+		var ext = s.get_numext
+		var trunk = strip_numext
+		var trval = trunk.to_i
+		if ext == "u8" then
+			return trval.to_b
+		else if ext == "" then
+			return trval
+		else
+			# Should not happen until supported by the compiler
```

Comment:
This is more of a hack right now to use Numeric than anything else, in the medium to long-term, this should be the replacement for `to_i`, when #1266 is properly addressed